frontend/packages/employee-frontend/src/components/attendances/AttendancePageWrapper.tsx

```diff
@@ -0,0 +1,319 @@
+import AttendanceGroupSelectorPage from './AttendanceGroupSelectorPage'
+import AttendanceComingPage from './AttendanceComingPage'
+import AttendancePresentPage from './AttendancePresentPage'
+import { isSuccess, Loading, Result } from '~api'
+import LocalDate from '@evaka/lib-common/src/local-date'
+import Title from '~components/shared/atoms/Title'
+import AttendanceDepartedPage from './AttendanceDepartedPage'
+import AttendanceAbsentPage from './AttendanceAbsentPage'
+import { ContentArea } from '~components/shared/layout/Container'
+import Button from '~components/shared/atoms/buttons/Button'
+import { DaycareGroup } from '~types/unit'
+import IconButton from '~components/shared/atoms/buttons/IconButton'
+import Colors from '~components/shared/Colors'
+import { faSearch, faTimes } from '~icon-set'
+import { DefaultMargins } from '~components/shared/layout/white-space'
+import { FreeTextSearch } from '~components/common/Filters'
+import { useDebounce } from '~utils/useDebounce'
+import AttendanceList from './AttendanceList'
+
+const Name = styled.div`
+  background: white;
+`
+
+const NoMarginTitle = styled(Title)`
+  margin-top: 0;
+  margin-bottom: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+
+  button {
+    margin-left: ${DefaultMargins.m};
+    color: ${Colors.greyscale.medium};
+  }
+`
+
+const NoMarginTitle2 = styled(Title)`
+  margin-top: 0;
+  margin-bottom: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  background: ${Colors.blues.primary};
+  color: ${Colors.greyscale.white};
+
+  button {
+    margin-left: ${DefaultMargins.m};
+    color: ${Colors.greyscale.white};
+  }
+`
+
+export const WideButton = styled(Button)`
+  width: 100%;
+`
+
+const Bold = styled.div`
+  font-weight: 600;
+`
+
+const SearchBar = animated(styled.div`
+  position: absolute;
+  background: ${Colors.greyscale.lightest};
+  width: 100vw;
+  overflow: hidden;
+  z-index: 1;
+`)
+
+export default React.memo(function AttendancePageWrapper() {
+  const { i18n } = useTranslation()
+  const { id, groupid } = useParams<{ id: string; groupid: string }>()
+  const location = useLocation()
+
+  const [unitData, setUnitData] = useState<Result<UnitData>>(Loading())
+  const [unit, setUnit] = useState<Result<UnitResponse>>(Loading())
+  const [groupAttendances, setGoupAttendances] = useState<
+    Result<ChildInGroup[]>
+  >(Loading())
+  const [showSearch, setShowSearch] = useState<boolean>(false)
+  const [freeText, setFreeText] = useState<string>('')
+  const [searchResults, setSearchResults] = useState<ChildInGroup[]>([])
+  const debouncedSearchTerms = useDebounce(freeText, 500)
+
+  useEffect(() => {
+    void getUnitData(id, LocalDate.today(), LocalDate.today()).then(setUnitData)
+    void getDaycare(id).then(setUnit)
+    groupid === 'all'
+      ? void getDaycareAttendances(id).then(setGoupAttendances)
+      : void getChildrenInGroup(groupid).then(setGoupAttendances)
```

Comment:
I wonder if we should only use `getDaycareAttendances` and ditch the other one. Fetching only a single group sometimes may have slightly better performance, but having only one DB-query and endpoint would simplify the code enough to possibly be a worthy tradeoff.